tests/test_transaction_db.py

```diff
@@ -729,21 +734,27 @@ def test_user_preferences(self):
         self.assertEqual(user_id_2, prefs['user_id'])
         self.assertEqual('de', prefs['report_language'])
 
-    @unittest.expectedFailure
+    # TODO
+    @unittest.skip("Key validation does not work for whatever reason.")
```

Comment:
I don't understand why this test is failing. It was even before I started working on the project and chaging things. It should get rejected by SQLAlchemy by default, right?

Reply:
for me it looks like the initial key validation was not properly implemented. 

here in `set_user_preferences`, it assumes that `setattr` will fail if trying to add a key that does not exist

Reply:
Yes. I thought `setattr` should fail as well, but it doesn't. Why not? I did some googling and couldn't find anything related. Maybe I wasn't searching right. Does one have to explicitly tell the `Model` that it cannot accept unknown arguments? How does one do that?

Reply:
mm, i dont think setattr will fail here, because it can either change an existing attribute, or add a new one. in this case it would just add a new attribute to the object

Reply:
one idea would be to add a check before setattr, and check that the attribute already exists for the object.